velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -869,8 +869,15 @@ struct ArrayRemoveFunction {
   void call(Out& out, const In& inputArray, E element) {
     for (const auto& item : inputArray) {
       if (item.has_value()) {
-        if (element != item.value()) {
-          out.push_back(item.value());
+        if constexpr (std::is_floating_point_v<E>) {
```

Comment:
Do we need to handle nan inside complex types? E.g., for array_remove(array(array(real)), array(real))?

Reply:
No, because complex elements end up using BaseVector::compare which already handles NaN. I have updated the PR with a unit test for the same.